com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs

```diff
@@ -927,6 +931,22 @@ public void OnAfterDeserialize()
                 k_AssetVersion = 7;
             }
 
+            if (k_AssetVersion < 8)
```

Comment:
Seems there was a merge issue at some point and we have ended up with 2 version 8 upgraders, this one needs to be 9 and moved after the version 8 below it

Reply:
and further down in static void UpgradeAsset(UniversalRenderPipelineAsset asset) we need to bump the last check to be version 9 not 8 too

Reply:
Thanks for noticing this. 

I fixed it in 1b2caec, can you please check the change?

Should I also bump the version numbers in [UpgradeAsset in the 2021.1](https://github.com/Unity-Technologies/Graphics/blob/e82aa3dc79a5e852bd88952530b9b46e83031ddd/com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs#L967) backport PR?

Reply:
Fix looks good, for now we have to hold off back porting to 21.1 since it seems there is another upgrader that hasn't been back ported, this would cause some severe issues with users so I'm looking to get answers on what to do here.

Reply:
Thanks for the info about 21.1. 
It seems that it would be better if both 2021.2 (this PR) and 2021.1 (backport PR) landed together at the same time?
Please notify when you know what more we should do for 21.1.

Reply:
@lukaschod 
Will https://github.com/Unity-Technologies/Graphics/pull/3089 be back-ported to branch https://github.com/Unity-Technologies/Graphics/tree/2021.1/staging ?

It seems that `UniversalRenderPipelineAsset.k_AssetVersion = 8` means "UniversalRenderPipelineAsset version containing improved shadow fade settings".

(With the current PR, we would have `UniversalRenderPipelineAsset.k_AssetVersion = 9` meaning "UniversalRenderPipelineAsset version containing auto-upgraded spot light shadow resolutions".)


Having "auto-upgraded spot light shadow resolutions" available in 2021.2 but not in 2021.1 might be surprising for users ("when upgrading project to 2021.1 nothing happened, but it changed when upgrading again to 2021.2!?"). Therefore I would like to land both pull requests #3831 and #3832 together.

@lukaschod 
If you plan to back-port #3089 to 2021.1, then #3832 has to wait (because asset update must happen in increasing version order).
If you don't plan to back-port #3089 to 2021.1, then on 2021.1 we can directly skip `UniversalRenderPipelineAsset.k_AssetVersion` from 7 to 9, and we will be able to land #3831 and #3832.
Please tell us, so that we can decide which path to take.

Reply:
Nop I do not plan to backport #3089. It has some core changes and I do not want to introduce that risk.

Reply:
Thanks a lot for the info Lukas.

I updated `UniversalRenderPipelineAsset.k_AssetVersion` in my [2021.1 backport pull request](https://github.com/Unity-Technologies/Graphics/pull/3832/commits/997afaea67392ccc4edcbca3012ecc07f909ecab) to match with this pull request (2021.2).